src/editor/EditorManager.js

```diff
@@ -831,6 +863,7 @@ define(function (require, exports, module) {
     exports.registerInlineEditProvider    = registerInlineEditProvider;
     exports.registerInlineDocsProvider    = registerInlineDocsProvider;
     exports.registerJumpToDefProvider     = registerJumpToDefProvider;
+    exports.registerProvider              = registerProvider;
```

Comment:
I'm generally confused why you're inventing a new registration process for providers.  The `EditorManager` already has one https://github.com/mozilla/brackets/blob/master/src/editor/EditorManager.js#L390-L424.  You should piggy-back on that.